website/project/decorators.py

```diff
@@ -41,16 +42,50 @@ def _kwargs_to_nodes(kwargs):
         raise HTTPError(http.NOT_FOUND)
     return parent, node
 
+
 def _inject_nodes(kwargs):
     kwargs['parent'], kwargs['node'] = _kwargs_to_nodes(kwargs)
 
-def must_be_valid_project(func):
+
+def must_be_valid_project(func=None, retractions_valid=False):
+    """ Ensures permissions to retractions are never implicitly granted. """
 
     # TODO: Check private link
+    def must_be_valid_project_inner(func):
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+
+            _inject_nodes(kwargs)
+
+            if not retractions_valid and getattr(kwargs['node'].retraction, 'is_retracted', False):
+                raise HTTPError(
+                    http.BAD_REQUEST,
+                    data=dict(message_long='Viewing retracted registrations is not permitted')
+                )
+            else:
+                return func(*args, **kwargs)
+
+        return wrapped
+
+    if func:
+        return must_be_valid_project_inner(func)
+
+    return must_be_valid_project_inner
+
+
+def must_be_public_registration(func):
 
     @functools.wraps(func)
     def wrapped(*args, **kwargs):
+
         _inject_nodes(kwargs)
+
+        node = kwargs['node'] or kwargs['parent']
```

Comment:
With @samchrisinger 's recent node refactor, the node should always be on `kwargs['node']`. Is that correct, Sam?